src/core/ext/transport/chttp2/transport/writing.cc

```diff
@@ -711,7 +711,15 @@ grpc_chttp2_begin_write_result grpc_chttp2_begin_write(
         // ref will be dropped at end of write
       }
     } else {
-      GRPC_CHTTP2_STREAM_UNREF(s, "chttp2_writing:no_write");
+      if (grpc_core::IsScheduleCancellationOverWriteEnabled() &&
+          t->outbuf.c_slice_buffer()->length > orig_len) {
+        // Some bytes belonging to the stream have been included in the
```

Comment:
I'm interested in how this happens... would be nice to believe that we don't need to keep the entire call around just to write the last few bytes.